internals/overlord/state/identities.go

```diff
@@ -47,28 +52,63 @@ type LocalIdentity struct {
 	UserID uint32
 }
 
+// BasicIdentity holds identity configuration specific to the "basic" type
+// (for HTTP basic authentication).
+type BasicIdentity struct {
+	Password string // Note: this is the sha512-crypt hashed password.
+}
+
+// This is used to ensure we send a well-formed identity Name.
+var identityNameRegexp = regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9_\-]*$`)
```

Comment:
I'm not sure how stringent we want to be with this check, but is it worth also worth forbidding that the name doesn't end in a `_` or `-`? For ease of implementation, I think it's probably pretty reasonable to assert that identity names must be `>=3` characters.

Also, is it worth enforcing either only `_`s or `-`s? If we don't express an opinion, this can easily get messy later. (I don't know whether the time has already passed for this, though)

As a bonus, is it possible to prevent ugly sequences like `--`/`__`? Probably cleanest to do this with a simple `strings.Contains(..., "--")` rather than making the regex hard to read

Reply:
After an internal discussion with Ben, we think this is not a security issue, so we will leave it as is: users should be allowed to use underscores or dashes, and maybe two-character names make sense, for example, "Ed".